src/CompositeReferenceFieldManager.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_content;
+
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Field\FieldDefinitionInterface;
+use Drupal\field\FieldConfigInterface;
+
+/**
+ * Manager class for composite reference fields.
+ *
+ * @package Drupal\oe_content
+ */
+class CompositeReferenceFieldManager implements CompositeReferenceFieldManagerInterface {
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * CompositeReferenceFieldManager constructor.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager.
+   */
+  public function __construct(EntityTypeManagerInterface $entity_type_manager) {
+    $this->entityTypeManager = $entity_type_manager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getReferencingEntities(EntityInterface $entity): array {
+    $referencing_entities = [];
+    // Getting all fields which could have references to the given entity.
+    $fields = $this->entityTypeManager->getStorage('field_config')->loadByProperties([
+      'field_type' => 'entity_reference',
+    ]);
+    $fields = array_merge($fields, $this->entityTypeManager->getStorage('field_config')->loadByProperties([
```

Comment:
Try loading via query->condition with IN to see if you can merge them into one query.

Reply:
It worked!